### ISSUE TITLE
[image_picker] On iOS, camera overlay is clickable through, user can interact with widgets behind the overlay

### ISSUE THREAD

**Issue by msw333** [flutter/flutter#28335]
Tested On iOS (iPhone X). 

When launching image picker (source.camera), the underneath screen and its buttons can be pushed. I experienced this behavior in my chat app, when I could open the  keyboard while image picker was running, and could reproduced it with the official example. When opening the camera and 'pushing' the underneath button (e.g. camera again), I can trigger it which leads the app to crash. 

https://www.youtube.com/watch?v=alVME2pXVyw

**Comment by cyanglaz** [comment]
@msw333 I tried with an iphoneX and couldn't reproduce the issue. Could you provide a mini example app to reproduce the issue and paste the code here? Also, could you do a `flutter doctor -v` and paste it here as well?

**Comment by msw333** [comment]
Sure. I can reproduce it with any code of imagepicker, source: camera). 

Such as: https://github.com/flutter/plugins/blob/master/packages/image_picker/example/lib/main.dart

1. Just open the official example and remember where all the elements/buttons  of the first screen are

2. Open image picker (source: camera). Now you see the camera to take a photo. 

3. Tab on that camera screen on the exact position, where the button of the underneath screen is located. 

4. The button is triggered (or the app crashes). You see the triggered button if you close the camera screen.

**Comment by cyanglaz** [comment]
@msw333 Could you paste your `flutter doctor -v` with the test device connected here. Also what is the image_picker version that you used?

**Comment by linsyorozuya** [comment]
Same problem.
Tested On iOS.

1.You can replace the button with textview ,make the textview frame as big as you can.
2.Open image picker (source: camera). Now you see the camera to take a photo.
3.Touch the empty content on image picker above the textview location.

finally you can see the textview begin active.

**Comment by theapphideaway** [comment]
Any updates on this? I'm experiencing the same issue

**Comment by darshankawar** [comment]
Verified this issue using `image_picker` plugin repo (latest version) on latest stable with iPhone 6s+ (iOS 14.3) and observed that the issue is replicable. 

1. Tap on Camera icon.
2. Once camera is in action, remember the position of all buttons (ex: video recording button) and tap just below the `switch camera` icon. 
3. You will see camera recording screen after tapping `Cancel` on current live camera screen.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.22.6, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.22.6 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 9b2d32b605 (3 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.9.3

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • sdk gphone x86 arm (mobile) • emulator-5554                            •
      android-x86 • Android 11 (API 30) (emulator)
    • Darshan's iphone (mobile)   • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios         • iOS 14.2

! Doctor found issues in 1 category.

```

</details>

**Comment by darshankawar** [comment]
Also verified the same on Android S10 and doesn't seem to occur on it.

**Comment by TahaTesser** [comment]
Moved from  https://github.com/flutter/flutter/issues/39218#issuecomment-701374970


1. Launch the app, tap `take a photo`,
2.Take a picture, click use photo and it returns back to the home page, don't try to click "use photo" or "retake", just try to tap the textfield (yes, it's invisible, but just try that)
3. The keyboard will bring up and cover the picture, then you can do nothing but close the application.

I can reproduce the issue 

![ezgif com-gif-maker (4)](https://user-images.githubusercontent.com/48603081/94688021-aa9fe180-034a-11eb-9149-ff6e2917fd44.gif)

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      appBar: AppBar(
        title: Text("DEMO"),
      ),
      body: Column(
        children: [
          FlatButton(
            child: Text("take photo"),
            onPressed: _takePhotoAction,
          ),
          Container(
            padding: const EdgeInsets.symmetric(horizontal: 15),
            margin: const EdgeInsets.all(20),
            child: TextField(
              maxLines: 20,
              style: TextStyle(
                color: Color(0xff333333),
                fontSize: 14,
              ),
              decoration: InputDecoration(
                border: InputBorder.none,
                hintText: "Input text",
              ),
            ),
            decoration: BoxDecoration(
              color: Colors.green,
              borderRadius: BorderRadius.circular(6.0),
            ),
          )
        ],
      ),
    );
  }

  void _takePhotoAction() async {
    final picker = ImagePicker();
    try {
      await picker.getImage(
        source: ImageSource.camera,
      );
    } catch (error) {
      print(error);
    }
  }
}

```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.22.0-10.0.pre.428, on Mac OS X 10.15.7 19H2
    x86_64, locale en-GB)
    • Flutter version 1.22.0-10.0.pre.428 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision a06f385145 (11 hours ago), 2020-09-29 21:37:04 -0400
    • Engine revision d18c3ea75c
    • Dart version 2.11.0 (build 2.11.0-178.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

[✓] Connected device (4 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS
      14.0.1
    • macOS (desktop)        • macos                     • darwin-x64     • Mac
      OS X 10.15.7 19H2 x86_64
    • Web Server (web)       • web-server                • web-javascript •
      Flutter Tools
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 85.0.4183.121

• No issues found!

```

</details> 


### Workaround
https://github.com/flutter/flutter/issues/39218#issuecomment-753497706

**Comment by febg11** [comment]
Any update on this? I am still experiencing it on 0.7.5+3

**Comment by amberfan** [comment]
Any update? I'm facing the same problem too

**Comment by oodavid** [comment]
_copied from https://github.com/flutter/flutter/issues/39218#issuecomment-753497706_

@febg11 @amberfan - I have a flutter-centric workaround that shoves a dialog between the camera and underlying app to act as a way to capture gestures. Feel free to use:

```dart
// Your function
Future<void> _pickImage() async {
  try {
    _openIOSBlockingModal(
      context: context,
      source: ImageSource.camera,
    );
    PickedFile pickedFile = await _picker.getImage(source: source);
  } catch(e) {
    // Handle errors...
  }
  _closeIOSBlockingModal(
    context: context,
  );
}

// TODO: Remove this workaround when this issue is resolved:
// https://github.com/flutter/flutter/issues/39218
bool _modalIsOpen = false;
void _openIOSBlockingModal({
  BuildContext context,
  ImageSource source,
}) {
  String platform = Platform.operatingSystem;
  if (!_modalIsOpen && source == ImageSource.camera && platform == 'ios') {
    _modalIsOpen = true;
    showGeneralDialog(
      context: context,
      pageBuilder: (
        BuildContext buildContext,
        Animation<double> animation,
        Animation<double> secondaryAnimation,
      ) {
        return Material(
          type: MaterialType.transparency,
          child: Stack(
            children: [
              Align(
                alignment: Alignment.topRight,
                child: Padding(
                  padding: EdgeInsets.all(16),
                  child: IconButton(
                    icon: Icon(
                      Icons.cancel,
                      size: 42,
                      color: Colors.black.withOpacity(0.3),
                    ),
                    onPressed: () {
                      _closeIOSBlockingModal(
                        context: context,
                      );
                    },
                  ),
                ),
              ),
              Align(
                alignment: Alignment.center,
                child: Text('Selecting Image'),
              ),
            ],
          ),
        );
      },
      barrierDismissible: false,
      barrierLabel: MaterialLocalizations.of(context).modalBarrierDismissLabel,
      barrierColor: Colors.white.withOpacity(0.7),
      transitionDuration: Duration.zero,
    );
  }
}

void _closeIOSBlockingModal({
  BuildContext context,
}) {
  if (_modalIsOpen) {
    Navigator.of(context, rootNavigator: true).pop();
    _modalIsOpen = false;
  }
}
```

**Comment by renefloor** [comment]
@stuartmorgan I can replicate it with the Flutter example app. It is easiest replicating it by removing the actions from the floating action buttons and replacing that with a print statement. It looks like I can interact with the FABs but not with the dialog. I cannot click on the dialog actions, but the dialog also does not go away when clicking next to it. When I clicked on the FAB once to open the dialog I cannot click on the FAB again. Maybe this is because the Flutter engine blocks the UI and waits with the navigation action while the app is not visible.

I wanted to know if it was due to how we open the camera overlay, So I copied all relevant code into a native iOS project with only 2 big buttons. I was not able to click on the buttons while the camera overlay was active, so it doesn't have anything to do with how the viewcontroller is presented.

This is the code of the native iOS example: https://github.com/renefloor/ios_camera_overlay_test/blob/main/cameratest/ViewController.m

This is a screenshot of the example. The button at the bottom is really big.
<img src="https://user-images.githubusercontent.com/15101411/122016261-2f3f6c80-cdc1-11eb-92fd-c4c4cef7c4ee.png" alt="screenshot example iOS" width="300">

I've found an other (old) open issue about this: #14720

There is not really anything we can do in the plugin about this, other than showing a dialog to block the whole Flutter view. I wonder if we should close this issue as duplicate. At least it's good if everybody gives a thumbs up (don't comment) on #14720.
